### ISSUE TITLE
Broken install with PyTorch 1.9.

### ISSUE THREAD

**Issue by bzapf** [LeviViana/torch_sampling#7]
Hi, 

I tried installing with PyTorch version 1.9.0. with `python setup.py build_ext --inplace --user` but it fails.

I think the relevant part of the output is
```
...

In file included from /path/torch_sampling/csrc/cuda/reservoir_sampling.cu:1:0:
/path/torch_sampling/csrc/cuda/reservoir_sampling.cuh:15:10: fatal error: THC/THCTensorRandom.h: No such file or directory
 #include <THC/THCTensorRandom.h>
          ^~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

... 
```
Issue #3 look similar to me. Do you have some suggestion on how to fix it? I can provide full output if needed.
Thanks in advance,
Bastian

**Comment by LeviViana** [comment]
Hey !

Would you like to propose a MR to fix it ? I think it's a good exercise if you want to learn a bit on pytorch internals.

These are the steps I'd follow:

- Take this line off,  and compile, it will probably bug
- If it bugs saying that some function is not declared, then take this function and look for its new implementation in pytorch's official codebase
- if another bug pops up, let me know

See you !

**Comment by bzapf** [comment]
Hi, thanks for your fast response! I'll look into it and get back to you.

**Comment by bzapf** [comment]
I believe to have solved it in #8
